### PR TITLE
added cheat sheet for evil-mode for emacs

### PR DESCRIPTION
**Pull request by kedarbramhe**
---
https://duck.co/ia/view/emacs_evil_mode_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@kedarbramhe Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/emacs-evil-mode.json [hunk 1]
```diff
@@ -0,0 +1,195 @@
+{
+    "id": "emacs_evil_mode_cheat_sheet",
+    "name": "Evil-Mode for Emacs",
+    "description": "Shortcuts for the Evil-mode of the Emacs editor",
+    "metadata": {
+        "sourceName": "GiHub",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/emacs-evil-mode.json]: I think you mean `GitHub`.

File: share/goodie/cheat_sheets/json/emacs-evil-mode.json [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        "evil mode emacs",
+        "evil-mode for emacs",
+        "evil mode for emacs",
+        "evil emacs mode"
+    ],
+    "section_order": [
+        "Editing Text",
+        "Window Navigation",
+        "Discovery"
+
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/emacs-evil-mode.json]: Get rid of this blank line.

File: share/goodie/cheat_sheets/json/emacs-evil-mode.json [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            },
+            {
+                "key": ":ack ENT",
+                "val": "search across files"
+            },
+            {
+                "key": ",f",
+                "val": "next search result"
+            },
+	    {
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/emacs-evil-mode.json]: Correct the indentation on this and those below.

File: share/goodie/cheat_sheets/json/emacs-evil-mode.json [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+                "key": ":describe-char",
+                "val": "describes the char underneath the cursor, allows you to change color theme from there"
+            },
+            {
+                "key": ":customize-face",
+                "val": "editor for changing all color themes use tab and shift+tab to navigate"
+            },
+	     {
+                "key": "F1 k KEYSTROKE",
+                "val": " press F1 followed by k, then enter a key stroke. You'll get information about what editor method was invoked"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/emacs-evil-mode.json]: Just use `Information about method invoked by KEYSTROKE` or similar.

File: share/goodie/cheat_sheets/json/emacs-evil-mode.json [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+            },
+	    {
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/emacs-evil-mode.json]: Again, indentation needs to be fixed.

**Comment by kedarbramhe** [comment]
@GuiltyDolphin Thanks for the review ... I will make those changes

File: share/goodie/cheat_sheets/json/emacs-evil-mode.json [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+            },
+            {
+                "key": "hjkl",
+                "val": "to navigate"
+            },
+            {
+                "key": "w",
+                "val": "to move forward by one word"
+            },
+	    {
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/emacs-evil-mode.json]: More indentation needs to be fixed ;)

**Comment by moollaza** [comment]
@kedarbramhe thanks again for all the hard work you've put into creating this Emacs Evil-Mode Cheat Sheet. You've done a great job so far! We at DuckDuckGo, and the DuckDuckHack community really appreciate you taking the time to contribute.

It looks like this is just about finished?  If you're stuck or need any help, I'd love to chat if you're around this week or next. Please message me on slack (@moollaza) or send me an email (moollaza@duckduckgo.com) if you'd be interested in talking. 

Thanks again!

**Comment by kedarbramhe** [comment]
Hi , sorry for such a late response . I did some changes . Please review.
Thanks

**Comment by moollaza** [comment]
> Hi , sorry for such a late response . I did some changes . Please review them.

Hey there! Thanks for making the changes, it looks like the tests are failing though because of your alias. Once that's fixed this can be deployed to Beta for testing

File: share/goodie/cheat_sheets/json/emacs-evil-mode.json [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+    "description": "Shortcuts for the Evil-mode of the Emacs editor",
+    "metadata": {
+        "sourceName": "GitHub",
+        "sourceUrl": "https://github.com/amirrajan/devbox/wiki/EMACS-Evil-Cheat-Sheet"
+    },
+    "template_type": "keyboard",
+    "aliases": [
+        "evil emacs",
+        "emacs evil-mode",
+        "emacs evil mode",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/emacs-evil-mode.json]: @kedarbramhe please remove this alias as it's unnecessary

**Comment by kedarbramhe** [comment]
hi @moollaza , thanks for pointing that out .

File: share/goodie/cheat_sheets/json/emacs-evil-mode.json [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+            {
+                "key": "w",
+                "val": "to move forward by one word"
+            },
+	    {
+                "key": "b",
+                "val": "to move back by one word"
+            },
+	    {
+                "key": "shift+[",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/emacs-evil-mode.json]: there should be no `+`, these should each be wrapped in `[]`:

`"key": "[Shift] [\\[]"`

File: share/goodie/cheat_sheets/json/emacs-evil-mode.json [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+                "val": "move up by new lines"
+            },
+	    {
+                "key": "shift+]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/emacs-evil-mode.json]: same here, remove the `+`

**Comment by moollaza** [comment]
@kedarbramhe this LGTM. I left a couple comments about the use of `+` which needs to be fixed.

**Comment by daxtheduck** [comment]
@kedarbramhe this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**